### ISSUE TITLE
Shell sidebar menu not navigable using arrow keys

### ISSUE THREAD

**Issue by benborgers** [carbon-design-system/ibm-security#704]
## Bug - Shell sidebar menu not navigable using arrow keys

The expanded sidebar menu in the application shell cannot be navigated using the up/down or left/right arrow keys. 

![image](https://user-images.githubusercontent.com/30215449/88846327-7f1a4280-d1b3-11ea-9ad9-93cf0a49a84b.png)

I believe this issue would be fixed within the `Toolbar` component: 
https://github.com/carbon-design-system/ibm-security/blob/ea7a8969920e0f440af105b91056de62169c599d/src/components/Toolbar/Toolbar.js#L2

**Expected behavior -**

Opening the sidebar and then pressing the arrow keys moves focus through the list. 

**Actual behavior -**

Using arrow keys scrolls the entire page. To focus those elements, you need to tab forward through part of the page until you reach the sidebar. 

### Optional information

**Version -** `1.21.0`

**Comment by SimonFinney** [comment]
AFAIK, I believe this focus tabbing is the current expected behaviour, at least as per original implementation - is there a recommendation or some guidance to consider otherwise?

**Comment by benborgers** [comment]
@SimonFinney I'm checking with the AVT testers who reported the issue on our end, I'll let you know what they come back with. Thanks!

**Comment by Deriohig** [comment]
@SimonFinney  Following up on the above conversation I believe they would like that the focus worked in one of these two ways. When a menu item is open, the tabbing goes into the open menu before the next higher navigation menu item. or 2) there is some other way to get into the menu item such as arrow key. Also I noticed that sub-menu items hidden by a caret are focused while tabbing even though they are not visible (screenshot).
![image](https://user-images.githubusercontent.com/7848154/90509351-1b02f280-e151-11ea-9a17-6232c330f9c3.png)

**Comment by SimonFinney** [comment]
@petervachon Any guidance on preference here?

**Comment by petervachon** [comment]
Okay, so it does seem a bit weird when tabbing through, the menu items go through sub-navigation without showing it within the UI, before going to the next main menu items. I believe the expected behavior would be to either auto-expand the menu list or wait for the user to hit enter, then tab through sub-navigation.

The other possible idea is to Deri's point of using the arrow keys to hit the sub-nav. 

What are accessibility best practices for this type of scenario?

**Comment by Deriohig** [comment]
@petervachon  Best practise is vague, they just state that you should be able to access the menu with tabbing or arrow keys. The fact you cannot access the hidden ones is an accessibility bug for sure. The excessive amount of tabbing is more of an annoyance than an outright bug.  For me, the most natural thing would be to hit the right arrow key and for it to jump into that menu item's submenu (if it is open), and left to jump out. But that is just a personal opinion, I'm not an avid keyboard navigator.

**Comment by petervachon** [comment]
Thanks for the clarification Deri. @SimonFinney is there someone on Carbon we can get guidance on for the arrow keys? We definitely need to fix the bug for the hidden ones (that should be a priority).

**Comment by SimonFinney** [comment]
We have an open PR to fix the sub-navigation issue here - https://github.com/carbon-design-system/ibm-security/pull/730

I suspect @dakahn might be able to provide a recommendation for how the keyboard interaction should behave otherwise?

**Comment by dakahn** [comment]
The arrow key interaction you're referring to is specific to a navigation menu (anti-)pattern based on a common misinterpretation of WAI-ARIA Authoring Practices Guidelines. We avoid using `role="menu"` and instead opt for the far simpler unordered list of links navigable via <kbd>tab</kbd> and <kbd>shift</kbd>+<kbd>tab</kbd>. This meets our sighted keyboard users AND our screen reader user's expectations without making either switch keyboard control contexts from web to operating system where you might expect to use more exotic keyboard controls for menu navigation like <kbd>shift</kbd>+<kbd>F10</kbd>, <kbd>home</kbd>, and <kbd>end</kbd>. 

For a deeper dive on the topic here's the Excellent Adrian Roselli article [Don't Use Aria Menu Roles for Site Nav](https://adrianroselli.com/2017/10/dont-use-aria-menu-roles-for-site-nav.html)

**Comment by SimonFinney** [comment]
Thanks for the guidance @dakahn ! That's great to have for future reference.

@Deriohig it feels like your first suggestion might be most applicable given the above - when a toolbar menu item is toggled, the tabbing navigates into the panel first before moving on to the next toolbar menu item.

**Comment by Deriohig** [comment]
@SimonFinney Sounds good to me

**Comment by wangc35** [comment]
Hey @SimonFinney Is there any update on the following suggestion:

- When a toolbar menu item is toggled, the tabbing navigates into the panel first before moving on to the next toolbar menu item? 

Looking at the comments above, I see the sub-nav issue was resolved here https://github.com/carbon-design-system/ibm-security/pull/730, but not the tabbing one. Thanks.

**Comment by SimonFinney** [comment]
Hey @wangc35 ! There's no update here, but I can work with anyone that's interesting in contributing this enhancement
